File: Euclid/Messages/Incoming/GETORDERINFO.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using System.Linq;

namespace Euclid.Messages.Incoming
{
    class GETORDERINFO : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (!player.Authenticated)
                return;

            request.Skip(1);

            string section = request.GetArgument(0);
            string id = request.GetArgument(1);
            string extra = request.GetArgument(2);

            CatalogueItem item = CatalogueManager.Instance.Offers.Where(x => x.Data.Id == id).FirstOrDefault();

            if (item == null || item.Data.Section != section)
            {
                if (id == "deal")
                {
                    //m_session.getServer().getLogger().info("Catalogue", "Article Page: " + section + " arcticleID: " + articleID + " " + extra);
                }
                else
                {
                    //m_session.getServer().getLogger().info("Catalogue", "Article Page: " + section + " arcticleID: " + articleID);
                }
                //m_session.systemMsg("Invalid catalogue article!");
            }
            else
            {
                var customData = string.Empty;

                if (item.Definition.Data.IsDecoration || item.Definition.Data.Sprite == "poster")
                    customData = extra;

                CatalogueManager.Instance.OrderHistory[player.Details.Id] = new CatalogueOrder(item, customData);
                player.Send(new ORDERINFO(item.Data.Id, item.Data.Price, customData, item.Definition.Data.Name));
            }
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/GETORDERINFO.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.catalogue.CatalogueItem;
import euclid.game.catalogue.CatalogueManager;
import euclid.game.catalogue.CatalogueOrder;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.ORDERINFO;
import euclid.network.streams.Request;

public class GETORDERINFO implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (!player.isAuthenticated()) {
            return;
        }

        request.skip(1);

        String section = request.getArgument(0);
        String id = request.getArgument(1);
        String extra = request.getArgument(2);

        CatalogueItem item = CatalogueManager.getInstance().getOffers().stream()
                .filter(x -> x.getData().getId().equals(id))
                .findFirst()
                .orElse(null);

        if (item == null || !item.getData().getSection().equals(section)) {
            // Invalid catalogue article
        } else {
            String customData = "";

            if (item.getDefinition().getData().isDecoration() || item.getDefinition().getData().getSprite().equals("poster")) {
                customData = extra;
            }

            CatalogueManager.getInstance().getOrderHistory().put(player.getDetails().getId(),
                    new CatalogueOrder(item, customData));
            player.send(new ORDERINFO(item.getData().getId(), item.getData().getPrice(),
                    customData, item.getDefinition().getData().getName()));
        }
    }
}
